project/api/models.py

```diff
@@ -131,7 +131,36 @@ class CachedData(models.Model):
     event_type = models.PositiveIntegerField(choices=EVENT_TYPES)
 
     def save(self, *args, **kwargs):  # pylint: disable=W0221
-        self.rank = self.wikidata_id * randint(0, 10)  # TODO: implement #8
+        url = "https://query.wikidata.org/sparql"
+        query = """
+        SELECT ?item ?outcoming ?sitelinks ?incoming WHERE {{
+            BIND(wd:Q{wid} AS ?item)
+            ?item wikibase:statements ?outcoming.
+            ?item wikibase:sitelinks ?sitelinks.
+            {{
+                SELECT (COUNT(?s) AS ?incoming) ?item WHERE {{
+                ?s ?p ?item.
+                _:b3 wikibase:directClaim ?p.
+                }}
+                GROUP BY ?item
+            }}
+            SERVICE wikibase:label {{ bd:serviceParam wikibase:language "en". }}
+        }}
+        """.format(
+            wid=self.wikidata_id
+        )
+        req = get(url, params={"format": "json", "query": query})
+
+        try:
+            data = req.json()["results"]["bindings"][0]
+
+            incoming = int(data["incoming"]["value"])
+            sitelinks = int(data["sitelinks"]["value"])
+            outcoming = int(data["outcoming"]["value"])
+            self.rank = incoming * sitelinks * outcoming
```

Comment:
I have few questions
 - what if one of the multipliers equals zero?
I am not sure, that it is fair `outcoming == 0 => rank = 0`
- how we should normalize data?
Later with separate request getting maximum as 100% and minimum as 0%?
- this will spawn 2 requests to wikidata for birth and death events of one person, right?
That is fine

Reply:
* True, though if one of the multipliers is 0 (which happens on occasion) the others tend to not be much greater.
* I'm not quite sure what you mean by this?
* Yes it will, see my PR on https://github.com/chronhq/microservices/pull/1 for an example of this.

Reply:
Ok, fine.
We'll see how it goes, need to collect some entities first and analyze their rating values. We will tune the equation later along with rating normalization

Reply:
@whirish I think Mikler's point 2 is that we should have a rank always between 0 and 1. @MiklerGM yes I think we can start with this, we know it's not perfect, but today I don't know how to improve this.

Reply:
Yes, sure, we already discussed this matter with Dan (@v1kun). He has some ideas about normalization, but first we should understand the range of rankings

Reply:
Yeah, normalization would be nice. 
Some simple methods like Z-Score or Minmax approach could work, but still they'll based on our multiplication, and it's not nice.
Instead of it, I'd use some linear combination (a*x1+b*x2+c*x3), where x1,x2,x3 is number of links, a,b,c - some weights, so 0-values won't bother us.